include/libtorrent/entry.hpp

```diff
@@ -95,8 +95,30 @@ namespace libtorrent
 			bool operator()(T1 const& rhs, T2 const& lhs) const
 			{ return rhs < lhs; }
 		};
-	};
+
+		template<class T>
+		class map_string : public std::map<std::string, T, aux::strview_less>
+		{
+		};
+#else
+		template<class T>
+		class map_string : public std::map<std::string, T>
+		{
+		private:
+			using base = std::map<std::string, T>;
+		public:
+			typename base::iterator find(const string_view& key)
```

Comment:
you should probably pull in the ``find()`` overloads from the base too: ``using base::find;``